### ISSUE TITLE
Editing Existing Embargo Date Does Not Persist

### ISSUE THREAD

**Issue by jeremyf** [samvera/hyrax#4534]
### Descriptive summary

In local development, with SHA ace33e13c0d7f851ed1c6c5939bbfa8e5d725b64, updating an existing embargo release date via the work's edit page does not persist the updated release date.

### Steps to reproduce the behavior

1. Create a work with a future embargo release date (7 days from now)
2. Edit the above created work, changing the future embargo release date to one day prior
3. Submit the form
4. Look at the works metadata, the embargo release date has not changed.

### Related work

I believe these are inter-related: #4517, #4516, #4515 

### Further Sleuthing

The related issues and this issue all route through https://github.com/samvera/hyrax/blob/ace33e13c0d7f851ed1c6c5939bbfa8e5d725b64/app/controllers/concerns/hyrax/works_controller_behavior.rb#L195-L198

Note the create action does correctly assign the initial given Embargo Release Date.

**Comment by jeremyf** [comment]
At the following, the `env.curation_concern.visibility_changed?` is set to true:

https://github.com/samvera/hyrax/blob/ace33e13c0d7f851ed1c6c5939bbfa8e5d725b64/app/actors/hyrax/actors/interpret_visibility_actor.rb#L214:L221

**Comment by jeremyf** [comment]
I believe I have tracked down what is happening:

Throughout the ActorStack, it appears that we apply the changed metadata correctly. In the following code, the `env.curation_concern.embargo` object shows the changed metadata.

https://github.com/samvera/hyrax/blob/ace33e13c0d7f851ed1c6c5939bbfa8e5d725b64/app/actors/hyrax/actors/base_actor.rb#L30-L34

When we debug at the following code, the `curation_concern.embargo` shows the updated value.  However, if I run `GenericWork.find(curation_concern.id).embargo` that object does not show the updated value.

 https://github.com/samvera/hyrax/blob/ace33e13c0d7f851ed1c6c5939bbfa8e5d725b64/app/controllers/concerns/hyrax/works_controller_behavior.rb#L100-L103

In other words, I believe that we are not applying cascade updates to the Embargo or Lease objects when they change.

**Comment by no-reply** [comment]
our current hypothesis is that this was introduced by some change in `hydra-head` or `active-fedora`. 

we're starting by getting an Actor Stack layer regression test in place.

**Comment by jeremyf** [comment]
In https://github.com/samvera/hyrax/blob/master/spec/features/actor_stack_spec.rb, I'm adding the following spec:

```ruby
  describe '#update' do
    context 'when changing embargo meta data' do
      let(:work) { create(:embargoed_work, with_embargo_attributes: { embargo_date: previous_embargo_date.to_s }) }
      let(:previous_embargo_date) { 7.days.from_now }
      let(:new_embargo_date) { 14.days.from_now }
      let(:attributes) { { "embargo_release_date" => new_embargo_date.to_s } }
      it 'updates the persistence layer for the curation concern' do
        expect do
          actor.update(env)
        end.to change { work.class.find(work.id).embargo_release_date.strftime("%Y-%m-%d") }.
                 from(previous_embargo_date.strftime("%Y-%m-%d")).
                 to(new_embargo_date.strftime("%Y-%m-%d"))
      end
    end
  end
```

This spec presently fails for the current head of the main branch.  I'm working on establishing a point to bisect.  This involves a lot of engine antics.

**Comment by jeremyf** [comment]
The spec fails in `v2.8.0` and `v2.9.0`; I'm heading further back in time.

**Comment by MariaStarPower** [comment]
Cannot test at this time because I am unable to create a new work.

![image](https://user-images.githubusercontent.com/32848057/101080084-2ee62f00-355d-11eb-9661-e6778b592020.png)

![image](https://user-images.githubusercontent.com/32848057/101080170-4a513a00-355d-11eb-8e4b-f22c401baf9b.png)

**Comment by crisr15** [comment]
Tested editing an existing embargo on a file set and the new date persisted.
